In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [55]:
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
import numpy as np

In [80]:
iris_dataset = load_iris()

In [81]:
print("Target names: {}". format(iris_dataset['target_names']))

Target names: ['setosa' 'versicolor' 'virginica']


In [6]:
print("Feature names: {}". format(iris_dataset['feature_names']))

Feature names:['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [7]:
print("Type of data: {}". format(type(iris_dataset['data'])))

Type of data: <class 'numpy.ndarray'>


In [8]:
print("shape of data: {}". format(iris_dataset['data'].shape))

shape of data: (150, 4)


In [9]:
print("Type of target: {}".format(type(iris_dataset['target'])))

Type of target: <class 'numpy.ndarray'>


In [10]:
print("Shape of target: {}".format(iris_dataset['target'].shape))

Shape of target: (150,)


In [11]:
print("Target:\n{}".format(iris_dataset['target']))

Target:
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


In [13]:
X_train, X_test, y_train, y_test = train_test_split(iris_dataset['data'], iris_dataset['target'], random_state=0)

In [14]:
print("X_train shape: {}". format(X_train.shape))

X_train shape: (112, 4)


In [15]:
print("y_train shape: {}". format(y_train.shape))

y_train shape: (112,)


In [18]:
print("X_test shape: {}".format(X_test.shape))

X_test shape: (38, 4)


In [19]:
print("y_test shape: {}". format(y_test.shape))

y_test shape: (38,)


In [20]:
knn = KNeighborsClassifier(n_neighbors=1)

In [22]:
knn.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=1)

In [23]:
X_new = np.array([[5, 2.9, 1, 0.2]])

In [24]:
print("X_new.shape: {}". format(X_new.shape))

X_new.shape: (1, 4)


In [31]:
prediction = knn.predict(X_new)
print("Prediction: {}".format(prediction))
print("Predicted target name: {}".format(iris_dataset['target_names'][prediction]))

Prediction: [0]
Predicted target name: ['setosa']


In [32]:
y_pred = knn.predict(X_test)
print("Test set predictions:\n {}".format(y_pred))
print("Test set score (np.mean): {:.2f}".format(np.mean(y_pred == y_test)))

Test set predictions:
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]
Test set score (np.mean): 0.97


In [33]:
print("Test set score (knn.score): {:.2f}".format(knn.score(X_test, y_test)))

Test set score (knn.score): 0.97


For this model, the accuracy on the test set is 0.97, which means the model made the right prediction for 97% of the irises in the given dataset. We can expect the model to be correct 97% of the time for predicting the species of new irises.
For a hobby botanist application, this is a high level of accuracy and it means that our model may be trustworthy enough to use.
Summary